db/sg_replicate_cfg.go

```diff
@@ -206,7 +212,7 @@ func (rc *ReplicationConfig) ValidateReplication(fromConfig bool) (err error) {
 		return base.HTTPErrorf(http.StatusBadRequest, "Replication remote URL is invalid")
 	}
 
-	if (remoteURL != nil && remoteURL.User.Username() != "") && rc.Username != "" {
+	if (remoteURL != nil && remoteURL.User.Username() != "") && rc.RemoteUsername != "" {
```

Comment:
It looks like this needs to be an _additional_ case. We want to limit the users to only providing credentials in one place. With this change it becomes possible to set in the deprecated `username` field and one of the above fields.

Reply:
Fixed that. Can also limit it so the `username` and `remote_username` fields cannot be used at the same time if needed. 
```suggestion
	if rc.RemoteUsername != "" && rc.Username != "" {
		return base.HTTPErrorf(http.StatusBadRequest,
			"Cannot set remote_username and username config options. Please only set the remote_username and remote_password config options. ")
	}

	if (remoteURL != nil && remoteURL.User.Username() != "") && (rc.RemoteUsername != "" || rc.Username != "") {
```